CODINGSTYLE.md

```diff
@@ -0,0 +1,443 @@
+* There must not be whitespace characters at the end of a line
+
+### Line length
+
+Lines should be limited at 80 characters.
+In some cases, if breaking a line makes it significantly less readable, it can go up to 120 characters.
+
+Rationale: when reviewing on GitHub, only 120 characters are visible; when reviewing on a mobile phone, only 80 characters are visible.
+And we do review on mobile phone quite a lot.
+
+#### Where to break a line
+
+A line can be broken after either a `,` or a `(`, or possibly after a binary operator in a long expression.
+
+### Indentation
+
+In general, indentation is 2 spaces, except continuation lines, which are indented 4 spaces.
+A continuation line is a line appearing because we broke something that should have been on one line for line length reasons.
+Typically, this means inside parentheses (formal or actual parameters of methods and constructors), and a long `extends` clause.
+
+Note that breaking a line right after the `=` sign of an initialization or assignment is *not* considered a continuation line.
+
+### Blank lines
+
+* Never put two blank lines in a row
+* (Almost) always put a blank line between two declarations in a class
+* Always put blank lines around a `case` whose body spans several lines
+* Insert blank lines at will between logical blocks of statements in a method
+
+The blank line between two consecutive declarations in a class can sometimes be omitted, if the declarations are single-line (which also means ScalaDocless) and strongly related.
+This happens pretty rarely (mostly a series of private fields).
+The rule of thumb is to always put a blank line.
+
+### Braces
+
+Braces should not be used for one-liner methods.
+Similarly, you should not use braces for the body of a `while`/`if`/`for` that is a one-liner.
+
+```scala
+def plus(x: Int, y: Int): Int =
+  x + y
+
+def clamp(v: Int, min: Int, max: Int): Int = {
+  if (v < min) min
+  else if (v > max) max
+  else v
+}
+```
+
+Every expression that spans more than one line should be enclosed in braces.
+For example, this is wrong:
+
+```scala
+val iterator =
+  new Iterator[A] {
+    ...
+  }
+```
+
+It should be:
+
+```scala
+val iterator = {
+  new Iterator[A] {
+    ...
+  }
+}
+```
+
+#### Two-liner if/else
+
+As an exception to the above rule, when a two-liner `if/else` (see section about `if/else` in general) is used as the right-hand-side of a definition or assignment, i.e., after an `=` sign, the braces around it can and should be omitted.
+For example:
+
+```scala
+def abs(x: Int): Int =
+  if (x >= 0) x
+  else -x
+```
+
+Note that the following formatting is not considered a two-liner if/else, and is therefore not valid:
+
+```scala
+def abs(x: Int): Int =
+  if (x >= 0)
+    x
+  else
+    -x
+```
+
+#### Braces in lambdas
+
+In lambdas (anonymous functions), the opening brace must be placed before the formal arguments, and not after the `=>`:
+
+```scala
+val f = { (x: Int) =>
+  body
+}
+```
```

Comment:
Almost. You should still avoid the braces around the body of the lambda, so it's this:
```scala
val someLongIdentifierWithHighIdentation = {  
  (x: Int) =>
    body
}
```